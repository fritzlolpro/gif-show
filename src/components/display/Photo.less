@iphones: ~"only screen and (max-width : 479px) and (min-width : 321px)";
@tablet: ~"only screen and (max-width : 1023px) and (min-width : 480px)";
.grid-photo-wrap {
    margin-bottom: 16px;
    .grid-photo {
        box-sizing: border-box;
        border: 16px solid #fff;
        border-radius: 5px;
        width: 100%;
        @media @iphones {
            border-width: 8px;
        }
    }
}