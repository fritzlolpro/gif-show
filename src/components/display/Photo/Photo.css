.grid-photo-wrap {
  margin-bottom: 16px;
}
.grid-photo-wrap .grid-photo {
  box-sizing: border-box;
  border: 16px solid #fff;
  border-radius: 5px;
  width: 100%;
}
@media only screen and (max-width : 479px) and (min-width : 321px) {
  .grid-photo-wrap .grid-photo {
    border-width: 8px;
  }
}
