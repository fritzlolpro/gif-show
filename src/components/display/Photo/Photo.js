import React from 'react'
import {Link} from 'react-router-dom'
import './Photo.css'
const Photo = ({
    imgLoad,
    photo,
    clickable = true
}) => {
    return (
        <div className='grid-photo-wrap'>
            {photo &&
            <Link
                to={clickable
                ? `../view/${photo.id}`
                : `#`}>
                <img
                    src={photo.images.downsized_medium.url}
                    alt={photo.title}
                    data-id={photo.id}
                    className='grid-photo'/>
            </Link>}
        </div>
    )
}
export default Photo