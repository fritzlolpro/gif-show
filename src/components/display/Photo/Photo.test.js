import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, {shallow, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

import Photo from './Photo'

describe('Grid', () => {

    it('Photo', () => {
        expect(Photo).toBeDefined();
    });

    it('Renders without props', () => {
        const tree = shallow(<Photo/>);
        expect(tree).toMatchSnapshot();
    });

    const props = {
        photos: [
            {
                id: '123',
                title: 'test title',
                images: {
                    downsized_medium: {
                        url: 'test_url'
                    }
                }
            }
        ]
    }
    it('Renders with props', () => {
        const tree = shallow(<Photo {...props}/>);
        expect(tree).toMatchSnapshot();
    });

})