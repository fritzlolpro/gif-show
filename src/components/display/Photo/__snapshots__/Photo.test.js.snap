// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Renders with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Photo
    photos={
        Array [
            Object {
              "id": "123",
              "images": Object {
                "downsized_medium": Object {
                  "url": "test_url",
                },
              },
              "title": "test title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "grid-photo-wrap",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "grid-photo-wrap",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Grid Renders without props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Photo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "grid-photo-wrap",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "grid-photo-wrap",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
